appstructure/appstructure.go

```diff
@@ -0,0 +1,187 @@
+		// LibraryId connects dimension to separately defined dimension
+		LibraryId string `json:"libraryId,omitempty"`
+		// LabelExpression optional parameter with label expression
+		LabelExpression string `json:"labelExpression,omitempty"`
+		// Defs definitions of dimension
+		Defs []string `json:"defs,omitempty"`
+		// Labels of dimension
+		Labels []string `json:"labels,omitempty"`
+	}
+
+	// AppStructureObject sense object structure
+	AppStructureObject struct {
+		AppObjectDef
+		MetaDef
+		// RawBaseProperties of Sense object
+		RawBaseProperties json.RawMessage `json:"rawBaseProperties,omitempty"`
+		// RawExtendedProperties of extended Sense object
+		RawExtendedProperties json.RawMessage `json:"rawExtendedProperties,omitempty"`
+		// RawGeneratedProperties inner generated properties of auto-chart
+		RawGeneratedProperties json.RawMessage `json:"rawGeneratedProperties,omitempty"`
+		// Children to the sense object
+		Children map[string]string `json:"children,omitempty"`
+		// Selectable true if select can be done in object
+		Selectable bool `json:"selectable"`
+		// Dimensions meta information of dimensions defined in object
+		Dimensions []AppStructureDimensionMeta `json:"dimensions,omitempty"`
+		// Measures meta information of measures defined in object
+		Measures []AppStructureMeasureMeta `json:"measures,omitempty"`
+		// ExtendsId ID of linked object
+		ExtendsId string `json:"extendsId,omitempty"`
+		// Visualization visualization of object, if exists
+		Visualization string `json:"visualization,omitempty"`
+	}
+
+	// AppStructureAppMeta meta information about the app
+	AppStructureAppMeta struct {
+		// Title of the app
+		Title string `json:"title"`
+		// Guid of the app
+		Guid string `json:"guid"`
+	}
+
+	// AppStructureBookmark list of bookmarks in the app
+	AppStructureBookmark struct {
+		// ID of bookmark
+		ID string `json:"id"`
+		// Title of bookmark
+		Title string `json:"title"`
+		// Description of bookmark
+		Description string `json:"description"`
+		// SheetId connected sheet ID, null if none
+		SheetId *string `json:"sheetId,omitempty"`
+		// SelectionFields fields bookmark would select in
+		SelectionFields string `json:"selectionFields"`
+		// RawProperties of Bookmark object
+		RawProperties json.RawMessage `json:"rawProperties,omitempty"`
+	}
+
+	// AppStructure of Sense app
+	AppStructure struct {
+		AppMeta AppStructureAppMeta `json:"meta"`
+		// Objects in Sense app
+		Objects map[string]AppStructureObject `json:"objects"`
+		// Bookmark list of bookmarks in the app
+		Bookmarks map[string]AppStructureBookmark `json:"bookmarks"`
+	}
+
+	// AppStructurePopulatedObjects is the type returned by an action when prompted for selectable objects
+	AppStructurePopulatedObjects struct {
+		// Parent id of the parent object
+		Parent string
+		// Objects first level app objects returned by the current action
+		Objects []AppStructureObject
+		// Sheets objects of sheet type
+		Sheets []AppStructureObject
```

Comment:
There's no reason to separate sheets, it's just an object with a hierarchy, same as e.g. filterpane.